backend/src/routes/generateClaimLink.ts

```diff
@@ -0,0 +1,54 @@
+import type { FastifyInstance } from 'fastify';
+import * as schema from '../db/schema';
+
+interface ClaimRequestBody {
+  amount: string;
+  address: string;
+  signature: string[];
+}
+
+export function getGenerateClaimLinkRoute(fastify: FastifyInstance): void {
+  fastify.post<{ Body: ClaimRequestBody }>(
+    '/generate_claim_link',
+    {
+      schema: {
+        body: {
+          type: 'object',
+          required: ['address', 'amount', 'signature'],
+          properties: {
+            amount: { type: 'string', pattern: '^[0-9]{1,78}.[0-9]{6}$' },
```

Comment:
IMHO requiring six `0` will result in the API consumers being annoyed at it. I would use something like `^([0-9]{1,78})(.[0-9]{1,6})?$` to make the decimals optional.

Reply:
indeed, made this constant to stay coherent with the db value but it's useless